src/glob.zig

```diff
@@ -501,6 +506,53 @@ pub fn GlobWalker_(
                     };
                 };
 
+                // Optimization:
+                // If we have a pattern like:
+                // `packages/*/package.json`
+                //              ^ and we are at this component, with let's say
+                //                a directory named: `packages/frontend/`
+                //
+                // Then we can just open `packages/frontend/package.json` without
+                // doing any iteration on the current directory.
+                //
+                // More generally, we can apply this optimization if we are on the
+                // last component and it is a literal with no special syntax.
+                if (component_idx == this.walker.patternComponents.items.len -| 1 and
+                    this.walker.patternComponents.items[component_idx].syntax_hint == .Literal)
+                {
+                    defer {
+                        this.closeDisallowingCwd(fd);
+                    }
+                    const pathz = try this.walker.arena.allocator().dupeZ(u8, this.walker.patternComponents.items[component_idx].patternSlice(this.walker.pattern));
```

Comment:
does this slice need to be allocated? looking into `prepareMatchedPath` it seems to be allocated again after joining